src/systemd.rs

```diff
@@ -23,6 +30,46 @@ pub fn stop_systemd_target() {
 	)
 }
 
+///Determine if systemd is used as the init system. This should work on all linux distributions.
+pub fn is_systemd_used() -> &'static bool {
+	static IS_SYSTEMD_USED: OnceLock<bool> = OnceLock::new();
+	IS_SYSTEMD_USED.get_or_init(
+		|| match Command::new("readlink").args(&["/sbin/init"]).output() {
+			Ok(output) => {
+				let init = String::from_utf8_lossy(&output.stdout);
+				init.trim().ends_with("/lib/systemd/systemd")
+			}
```

Comment:
Wouldn't it be enough to simply check, if the dbus interface exists? In other words, just don't treat an error of `spawn_scope` as fatal. Log it and continue operation.

Reply:
I do think we should have some sort of boolean check before attempting. Cosmic might be packaged for a distribution with systemd support, but the end user might not use systemd as their init system (for whatever reason).

In that case, trying and then logging errors is just spamming irrelevant messages for their system.

Reply:
plus checking a rwlocked bool is probably cheaper than failing to execute a dbus call, for whatever those few clock cycles are worth

Reply:
Worked on this a little bit. Probably bit twiddling but I'm having trouble doing this without using async, or calling a blocking variant(which zbus docs say may deadlock a thread). Does zbus provide a way of checking for existence of an interface without connecting? I can make the function async but `if *systemd_is_used.await` just kills me a little inside.

Reply:
@Drakulix so systemctl will throw an error if systemd is not the init system([relevant SO question](https://askubuntu.com/questions/1379425/system-has-not-been-booted-with-systemd-as-init-system-pid-1-cant-operate), do you think it's safe to assume that if `systemctl -v` errors out for any reason, that we should assume systemd is not used

Reply:
so I switched to a different method, which it seems is used by [libsystemd](https://manpages.ubuntu.com/manpages/xenial/en/man3/sd_booted.3.html#NOTES:~:text=Internally%2C%20this%20function%20checks%20whether). If you would rather use another method, I have a few other things I could try (like filtering `cat /proc/1/stat`)

Reply:
seems good to me.